tuxemon/event/actions/random_battle.py

```diff
@@ -0,0 +1,157 @@
+#
+# Tuxemon
+# Copyright (c) 2014-2017 William Edwards <shadowapex@gmail.com>,
+#                         Benjamin Bean <superman2k5@gmail.com>
+#
+# This file is part of Tuxemon
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import annotations
+from tuxemon import monster
+
+import json, random, tuxemon, logging
+import tuxemon.npc
+from tuxemon import ai
+from tuxemon.db import db
+
+from tuxemon.combat import check_battle_legal
+from tuxemon.event.eventaction import EventAction
+from typing import Union, NamedTuple, final, Optional
+from os import listdir
+from tuxemon.states.world.worldstate import WorldState
+from tuxemon.states.combat.combat import CombatState
+class AddMonsterActionParameters(NamedTuple):
+
+    monster_level: int
+    npc_slug: str
+    tile_pos_x: int
+    tile_pos_y: int
+    animations: Union[str, None]
+    behavior: Union[str, None]
+
+
+logger = logging.getLogger(__name__)
+@final
+class randomfight(EventAction[AddMonsterActionParameters]):
+    """
+    a
```

Comment:
Please be more descriptive.  and include what this does, specifically, and why it is difference than a "random encounter"